idiss-csharp/IdissLib/Idiss.cs

```diff
@@ -22,25 +23,28 @@ public static class Idiss
         UInt64 expiry,
         [MarshalAs(UnmanagedType.LPArray)] byte[] ip_private_key, int ip_private_key_ptr_len,
         [MarshalAs(UnmanagedType.LPArray)] byte[] ip_cdi_private_key, int ip_cdi_private_key_ptr_len,
-        out int out_length, out int out_success);
+        out int out_length, out int out_capacity, out int out_success);
 
         [DllImport("idiss.dll")]
         private static extern IntPtr validate_request_v1_cs([MarshalAs(UnmanagedType.LPArray)] byte[] ctx, int ctx_len,
         [MarshalAs(UnmanagedType.LPArray)] byte[] ip_info, int ip_info_len,
         [MarshalAs(UnmanagedType.LPArray)] byte[] ars_infos, int ars_infos_len,
-        [MarshalAs(UnmanagedType.LPArray)] byte[] request, int request_len, out int out_length);
+        [MarshalAs(UnmanagedType.LPArray)] byte[] request, int request_len, out int out_length, out int out_capacity);
 
         [DllImport("idiss.dll")]
         private static extern IntPtr create_identity_object_v1_cs([MarshalAs(UnmanagedType.LPArray)] byte[] ip_info, int ip_info_len,
         [MarshalAs(UnmanagedType.LPArray)] byte[] request, int request_len,
         [MarshalAs(UnmanagedType.LPArray)] byte[] alist, int alist_len,
         [MarshalAs(UnmanagedType.LPArray)] byte[] ip_private_key, int ip_private_key_ptr_len,
-        out int out_length, out int out_success);
+        out int out_length, out int out_capacity, out int out_success);
 
         [DllImport("idiss.dll")]
         private static extern IntPtr validate_recovery_request_cs([MarshalAs(UnmanagedType.LPArray)] byte[] ctx, int ctx_len,
         [MarshalAs(UnmanagedType.LPArray)] byte[] ip_info, int ip_info_len,
-        [MarshalAs(UnmanagedType.LPArray)] byte[] request, int request_len, out int out_length);
+        [MarshalAs(UnmanagedType.LPArray)] byte[] request, int request_len, out int out_length, out int out_capacity);
+        
+        [DllImport("idiss.dll")]
+        private static extern IntPtr free_array_len_cap(IntPtr array_ptr,  uint out_length, uint out_capacity);
```

Comment:
The `out_length` and `out_capacity` needs to be `ulong` to [match](https://learn.microsoft.com/en-us/dotnet/standard/native-interop/type-marshalling#default-rules-for-marshalling-common-types) the rust side

```suggestion
        private static extern IntPtr free_array_len_cap(IntPtr array_ptr, ulong out_length, ulong out_capacity);
```